### PR TITLE
Add framework package

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2440 and https://github.com/giantswarm/giantswarm/issues/2441

Extracted code from aws-operator e2e tests, this can be reused in other projects.

### CODE REVIEW

File: Gopkg.toml [hunk 1]
```diff
@@ -41,6 +45,11 @@
   branch = "master"
   name = "github.com/spf13/cobra"
 
+[[constraint]]
+  branch = "master"
+  name = "github.com/cenkalti/backoff"
+
+
```
Comment (xh3b4sd) [inline Gopkg.toml]: I would like to sort this alphabetically. There is also an additional empty line.

File: pkg/framework/error.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package framework
+
+import "github.com/giantswarm/microerror"
+
+var waitTimeoutError = microerror.New("waitTimeout")
```
Comment (xh3b4sd) [inline pkg/framework/error.go]: Same thing with the order here.

File: Gopkg.toml (hunk 1 above)
Comment (fgimenez) [reply Gopkg.toml]: Sure done.

File: pkg/framework/error.go (hunk 2 above)
Comment (fgimenez) [reply pkg/framework/error.go]: Done.

**Pull request merged**